src/gui/XournalppCursor.cpp

```diff
@@ -118,119 +132,127 @@ void XournalppCursor::updateCursor()
-
+	bool useDefault = false;
+	
+	
 	if (this->busy)
 	{
-		cursor = gdk_cursor_new_for_display(gdk_window_get_display(window), GDK_WATCH);
+		setStockCursor(GDK_WATCH);
 	}
 	else
 	{
 		ToolHandler* handler = control->getToolHandler();
 		ToolType type = handler->getToolType();
 
+
 		if (type == TOOL_HAND)
 		{
 			if (this->mouseDown)
 			{
-				cursor = gdk_cursor_new_for_display(gdk_window_get_display(window), GDK_FLEUR);
+	
+				setStockCursor(GDK_FLEUR);
 			}
 			else
 			{
-				cursor = gdk_cursor_new_for_display(gdk_window_get_display(window), GDK_HAND1);
+				setStockCursor(GDK_HAND1);
 			}
 		}
 		else if (!this->insidePage)
 		{
-			// not inside page: so use default cursor
+			useDefault = true; // not inside page: so use default cursor
 		}
 		else if (this->selectionType)
 		{
 			switch (this->selectionType)
 			{
 			case CURSOR_SELECTION_MOVE:
-				cursor = gdk_cursor_new_for_display(gdk_window_get_display(window), GDK_FLEUR);
+				setStockCursor(GDK_FLEUR);
 				break;
 			case CURSOR_SELECTION_TOP_LEFT:
-				cursor = gdk_cursor_new_for_display(gdk_window_get_display(window), GDK_TOP_LEFT_CORNER);
+				setStockCursor(GDK_TOP_LEFT_CORNER);
 				break;
 			case CURSOR_SELECTION_TOP_RIGHT:
-				cursor = gdk_cursor_new_for_display(gdk_window_get_display(window), GDK_TOP_RIGHT_CORNER);
+				setStockCursor(GDK_TOP_RIGHT_CORNER);
 				break;
 			case CURSOR_SELECTION_BOTTOM_LEFT:
-				cursor = gdk_cursor_new_for_display(gdk_window_get_display(window), GDK_BOTTOM_LEFT_CORNER);
+				setStockCursor(GDK_BOTTOM_LEFT_CORNER);
 				break;
 			case CURSOR_SELECTION_BOTTOM_RIGHT:
-				cursor = gdk_cursor_new_for_display(gdk_window_get_display(window), GDK_BOTTOM_RIGHT_CORNER);
+				setStockCursor(GDK_BOTTOM_RIGHT_CORNER);
 				break;
 			case CURSOR_SELECTION_LEFT:
 			case CURSOR_SELECTION_RIGHT:
-				cursor = gdk_cursor_new_for_display(gdk_window_get_display(window), GDK_SB_H_DOUBLE_ARROW);
+				setStockCursor(GDK_SB_H_DOUBLE_ARROW);
 				break;
 			case CURSOR_SELECTION_ROTATE:
-				cursor = gdk_cursor_new_for_display(gdk_window_get_display(window), GDK_EXCHANGE);
+				setStockCursor(GDK_EXCHANGE);
 				break;
 			case CURSOR_SELECTION_DELETE:
-				cursor = gdk_cursor_new_for_display(gdk_window_get_display(window), GDK_PIRATE);
+				setStockCursor(GDK_PIRATE);
 				break;				
 			case CURSOR_SELECTION_TOP:
 			case CURSOR_SELECTION_BOTTOM:
-				cursor = gdk_cursor_new_for_display(gdk_window_get_display(window), GDK_SB_V_DOUBLE_ARROW);
+				setStockCursor(GDK_SB_V_DOUBLE_ARROW);
 				break;
 			default:
 				break;
 			}
-		}
-		else if (type == TOOL_PEN)
+		}		
+		else if ( type == TOOL_PEN || type == TOOL_HILIGHTER )
 		{
-			cursor = getPenCursor();
-
+			if ( this->inputDevice == INPUT_DEVICE_MOUSE && !this->mouseDown ) //mouse and not pressed
+			{
+				setStockCursor(GDK_ARROW);
+			}
+			else
+			{
+				if (type == TOOL_PEN)
+				{
+					cursor = getPenCursor();
+				}
+				else // must be:  if (type == TOOL_HILIGHTER)
```

Comment:
Either add an assertion here or keep the highlighter tool check; this assumption may not be true in the future.